src/App.tsx

```diff
@@ -15,36 +15,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'default',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const getPreparedGoods = (
+  goods: string[],
+  sortField: SortType,
+  isReverse: boolean,
+): string[] => {
+  const sortedGoods = [...goods].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isReverse ? sortedGoods.reverse() : sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.DEFAULT);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const goods = getPreparedGoods(goodsFromServer, sortField, isReverse);
+
+  const handleReset = () => {
+    setSortField(SortType.DEFAULT);
+    setIsReverse(false);
+  };
+
+  const isModified = sortField !== SortType.DEFAULT || isReverse;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortField !== SortType.ALPHABET ? 'is-light' : ''}`}
+          onClick={() => setSortField(SortType.ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${sortField !== SortType.LENGTH ? 'is-light' : ''}`}
+          onClick={() => setSortField(SortType.LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${!isReverse ? 'is-light' : ''}`}
+          onClick={() => setIsReverse(!isReverse)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isModified && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {goods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
Consider using a more unique key for the list items if possible. Using the item itself as a key can lead to issues if there are duplicate items in the list. If the items are guaranteed to be unique, this is fine, but otherwise, consider using an index or another unique identifier.